src/app/components/country-selector/country-selector.component.ts

```diff
@@ -0,0 +1,54 @@
+import { CountryService } from '../../services/country.service';
+import { Component, OnInit, ViewChild } from '@angular/core';
+import { CountryI } from '../../Interface/country';
+import { MatMenuTrigger } from '@angular/material/menu';
+
+@Component({
+  selector: 'app-country-selector',
+  templateUrl: './country-selector.component.html',
+  styleUrls: ['./country-selector.component.scss'],
+})
+export class CountrySelectorComponent implements OnInit {
+  country!: string;
+  seachValue!: string;
+  menuOpen: boolean = false;
+
+  allCountrys: CountryI[] = [
+    { alpha2Code: 'us', name: 'United States' },
+    { alpha2Code: 'de', name: 'Germany' },
+  ];
+
+  filteredCountrys!: CountryI[];
+
+  constructor(public countryService: CountryService) {}
+
+  @ViewChild('ClickMatMenuTrigger') trigger!: MatMenuTrigger;
+
+  ngOnInit(): void {
+    this.country = this.countryService.country;
+    this.filteredCountrys = this.allCountrys;
+  }
+
+  filterCountrys() {
+    this.filteredCountrys = this.allCountrys.filter((element) => {
+      return element.name
+        .toLocaleLowerCase()
+        .includes(this.seachValue.toLocaleLowerCase());
+    });
+  }
+
+  stopPropagation(event: any) {
+    event.stopPropagation();
+  }
```

Comment:
This looks unnecessary. Can we use `$event.stopPropagation` in the html template instead?

Reply:
function was removed and $event.stopPropagation() was added in the html